transformer_engine/pytorch/csrc/comm_gemm_overlap.h

```diff
@@ -207,7 +207,7 @@ struct UbufCommOverlap : torch::CustomClassHolder, UbufBase {
 
     // Set the number of SMs for GEMM with margin
     cudaDeviceProp prop;
-    cudaGetDeviceProperties(&prop, 0);
+    NVTE_CHECK_CUDA(cudaGetDeviceProperties(&prop, 0));
```

Comment:
This call is expensive. Is this constructor running only once or during every communication? In the layernorm we take that info from pyTorch because it caches that information. We should cache it as well.

Reply:
The constructor is executed only once per overlap type (e.g. "qkv_fprop", "fc1_dgrad", etc...) when we call `te.module.base.initialize_ub()`.

It's still not a bad idea for our own `cuda_driver.h` in TE/common to provide a cached version of this call. I'll add it to the PR.

Reply:
@denera `at::cuda::getCurrentDeviceProperties()` could be used, framework already caches these.

Reply:
I just replaced this chunk of code with `transformer_engine::cuda::sm_count()`, which is already a cached implementation, instead of manually getting it from device props.